client_test.go

```diff
@@ -133,8 +133,11 @@ func TestMain(m *testing.M) {
 			}
 		}
 		if err := os.RemoveAll(defaultRoot); err != nil {
-			fmt.Fprintln(os.Stderr, "failed to remove test root dir", err)
-			os.Exit(1)
+			// this directory is a volume for container tests, so we don't care if it fails
```

Comment:
Let's create the root within the volume and remove this code

Reply:
I think we can just remove this code anyway, we don't need to change the path.  

There's no reason to `os.Exit(1)` if the RemoveAll fails, just ignore the error in all cases. We already do that in dozens of places in the test suite.